R/scan_file.R

```diff
@@ -0,0 +1,32 @@
+#' Scan a px cube file and return px data and metadata
+#'
+#' @param px_file px cube file
+#'
+#' @return list with the data as a tibble and the metadata as a list
+#' @export
+#'
+#' @examples scan_px_file("px-x-0102020203_110.px")
+scan_px_file <- function (px_file) {
+  scanned_lines <- scan(px_file, what = "list", sep = ";", quote = NULL,
+                        quiet = TRUE, encoding = "latin1", multi.line = TRUE)
+  px_rows <- list()
+  lines <- c()
+  for (line in scanned_lines) {
+    if (line != "") {
+      lines <- append(lines, line)
+    } else {
+      if (length(lines) > 0) {
+        px_rows <- append(px_rows, get_keyword_values_pair(lines))
+        lines <- c()
+      }
+    }
+  }
+  data <- vectorize_px_data(lines)
+  metadata <- process_px_metadata(px_rows)
+  df <- expand.grid(c(metadata$HEADING, metadata$STUB))
+  df[, 'DATA'] = data
+  output <- list('metadata' = metadata,
+                 'tdf' = tibble::as_tibble(df))
```

Comment:
One nomenclature question: why `tdf` instead of just `df` ?
I would go for `df` or even `dataframe` (which I think is ok, as the data.frame format is being taken over in general by tibble)